underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java

```diff
@@ -261,7 +273,92 @@ public boolean mkdirs(String path, boolean createParent) throws IOException {
 
   @Override
   public boolean mkdirs(String path, MkdirsOptions options) throws IOException {
-    return true;
+    if (path == null) {
+      return false;
+    }
+    if (isDirectory(path)) {
+      return true;
+    }
+    if (isFile(path)) {
+      LOG.error("Cannot create directory {} because it is already a file.", path);
+      return false;
+    }
+    if (!options.getCreateParent()) {
```

Comment:
I feel the logic here is a bit complicated to reason, how about the following (equalavent way):
```java
if (!parentExists(path)) {
  if (!options.getCreateParent()) {
    LOG.error("Cannot create directory {} because parent does not exist", path);
    return false;
  }
  String parentKey = getParentKey(path);
  // Recursively make the parent folders
  if (!mkdirs(parentKey, true)) {
    LOG.error(...);
    return false;
  }
}
return mkdirsInternal(path);

```